lib/Zonemaster/Backend/Config.pm

```diff
@@ -15,27 +15,53 @@ use Readonly;
 use Zonemaster::Backend::Validator qw( :untaint );
 use Zonemaster::Backend::DB;
 
-our $path;
+Readonly my @SIG_NAME => split ' ', $Config{sig_name};
 
-if ($ENV{ZONEMASTER_BACKEND_CONFIG_FILE}) {
-    $path = $ENV{ZONEMASTER_BACKEND_CONFIG_FILE};
-}
-else {
-    my @search_paths = (
-        '/etc/zonemaster/backend_config.ini',
-        '/usr/local/etc/zonemaster/backend_config.ini',
-        dist_file('Zonemaster-Backend', "backend_config.ini")
-    );
+=head1 STATIC METHODS
 
-    for my $default_path (@search_paths) {
-        if ( -e $default_path ) {
-            $path = $default_path;
-            last;
-        }
-    }
-}
+=head2 get_default_path
 
-Readonly my @SIG_NAME => split ' ', $Config{sig_name};
+Determine the path for the default backend_config.ini file.
+A list of values and locations are checked and the first match is returned.
+If all places are checked and no file is found, an exception is thrown.
+
+This procedure is idempotent - i.e. if you call this procedure multiple times
+the same value is returned no matter if environment variables or the file system
+have changed.
+
+The following checks are made in order:
+
+=over 4
+
+=item $ZONEMASTER_BACKEND_CONFIG_FILE
+
+If this environment variable is set ot a truthy value, that path is returned.
+
+=item /etc/zonemaster/backend_config.ini
+
+If a file exists at this path, it is returned.
+
+=item /usr/local/etc/zonemaster/backend_config.ini
+
+If a file exists at such a path, it is returned.
+
+=item DIST_DIR/backend_config.ini
+
+If a file exists at this path, it is returned.
+DIST_DIR is wherever File::ShareDir installs the Zonemaster-Backend dist.
+
+=back
+
+=cut
+
+sub get_default_path {
+    state $path =
```

Comment:
I cannot see that `use feature state` is set. Isn't that necessary?

Reply:
It's enabled by default since Perl 5.10 and we're declaring a minimum version of 5.14.